patches/tModLoader/Terraria/ModLoader/Mod.cs

```diff
@@ -111,38 +111,15 @@ public void AddConfig(string name, ModConfig mc)
 
 		public void AddContent(ILoadable instance){
 			if (!loading)
-				throw new Exception("AddContent can only be called from Mod.Load");
+				throw new Exception(Language.GetTextValue("tModLoader.LoadErrorNotLoading"));
 			instance.Load(this);
-			loadables.Add(instance);
+			content.Add(instance);
+			ContentInstance.Register(instance);
 		}
 
-		/// <summary>
-		/// Gets the ModItem instance corresponding to the name. Because this method is in the Mod class, conflicts between mods are avoided. Returns null if no ModItem with the given name is found.
-		/// </summary>
-		/// <param name="name">The name.</param>
-		/// <returns></returns>
-		public ModItem GetItem(string name) => items.TryGetValue(name, out var item) ? item : null;
-
-		/// <summary>
-		/// Gets the internal ID / type of the ModItem corresponding to the name. Returns 0 if no ModItem with the given name is found.
-		/// </summary>
-		/// <param name="name">The name.</param>
-		/// <returns></returns>
-		public int ItemType(string name) => GetItem(name)?.item.type ?? 0;
-
-		/// <summary>
-		/// Gets the GlobalItem instance with the given name from this mod.
-		/// </summary>
-		/// <param name="name">The name.</param>
-		/// <returns></returns>
-		public GlobalItem GetGlobalItem(string name) => globalItems.TryGetValue(name, out var globalItem) ? globalItem : null;
+		public IEnumerable<ILoadable> GetContent() => content;
```

Comment:
May wanna make a readonly list field wrapping around content to return it here (still as IEnumerable.)

Reply:
I normally choose performance over safety in tML. I don't think this is a high traffic method so it's unlikely to matter and in general I like the idea of returning read only collections, but anyone re-casting it could also get the impl via reflection anyway.